spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/intermediate/IRToDnn.scala

```diff
@@ -115,19 +115,54 @@ private[bigdl] class IRToDnn extends ConvertBase[IRElement[Float], Module[Float]
 
   private def fromSpatialConvolution(node: IRElement[Float]) : Module[Float] = {
     val t = node.getOp().asInstanceOf[IRSpatialConvolution[Float]]
-    require(t.format == DataFormat.NCHW, "Dnn SpatialConvolution only supports NCHW")
-    ReflectionUtils.reflectFromIR(node, Class.forName(prefix + "SpatialConvolution"))
+    if (t.format == DataFormat.NCHW) {
+      ReflectionUtils.reflectFromIR(node, Class.forName(prefix + "SpatialConvolution"))
+    } else {
+      // special process for NHWC
+      require(t.nGroup == 1, "Only support nGroup is 1 for NHWC")
+      val layer = ReflectionUtils.reflectFromIR(node, Class.forName(prefix + "SpatialConvolution"))
+      val p = layer.parameters()
+      val weight = p._1(0)
+      val gradWeight = p._2(0)
+
+      val weightSize = weight.size() // for nchw
+      val newSize = Array(weightSize(2), weightSize(3), weightSize(1), weightSize(0))// for nhwc
```

Comment:
If the data format is `NHWC`,  the weight shape of conv is also `iohw`. Is it right ?

Reply:
Yes, weight shape of blas is `hwio`. But for dnn conv, it can still be `oihw` as input and weight format are `any`, and `reorder` will be done.